Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ReviewRequestController.java

```diff
@@ -21,12 +27,39 @@
            
     @RequestMapping(value = "/auth/request/add", method = RequestMethod.POST )
     public @ResponseBody ReviewRequest addNewReviewRequest( @ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
-  
-        reviewRequest.setStatus(ReviewStatus.OPENED);
         reviewRequest.setRequester(principal.getName());
-        
+        reviewRequest.setCreatedAt(LocalDateTime.now());
         requestsMgr.addReviewRequest(reviewRequest);
         
         return reviewRequest;
     }
+    
+    @RequestMapping(value = "/auth/request/resolve", method = RequestMethod.POST )
+    public @ResponseBody ResponseEntity<ReviewRequest> resolveRequest(@RequestBody ReviewRequest reviewRequest,Principal principal) {
+        if(reviewRequest.getStatus() != ReviewStatus.RESOLVED || reviewRequest.getComments().size() != 1) {
```

Comment:
as soon as we have functionality that allows to just add a comment to a request, this method will no longer work.
This method should take a review request id, status, and comment and if the review request for the id is not yet resolved, resolve it (set resolver, resolution time, add the comment, set the status).

Reply:
Shouldn't we have a separate endpoint when it comes to adding only comment ? 

Correct me if I'm wrong. In the above case, we are assuming that the first comment is tightly coupled with Resolving/Reopening the review request and the later coming comments are added without any status change.

But what if the user just need to update the comment in the beginning without any status change ?

Reply:
Yes, there will be a separate endpoint for adding comments. However, the reviewRequest is the object that can be commented on before it's being resolved, so 
1. a user creates a review request (with a note why it needs to be reviewed)
2. another user might make a comment about how it should be changed
3. the first user might make anotehr comment in response
4. step 3 and 4 might repeat or another user might make a comment
5. eventually a user will resolve the request; the request will have multiple comments by then.

Reply:
editing a comment will also be a different endpoint if we allow that.

Reply:
In Step 5: There is the possibility of having 0 or 1 comment only right in the request while resolving ? Because rest of the comments will already get added by separate end points. We must be sure that the reviewRequest coming to this method should not be more that 1.

One more clarification, if I'm getting a status other that "RESOLVED", you want me to update with the correct status instead of sending 400 ?

Reply:
Simply don't use the review request object here. It doesn't make sense to use incomplete review request objects, when all you need is the comment and the review request id. It's just confusing. If you want to use an object, create a dedicated class for the information, otherwise you can get the id from the url and the comment as request parameter. Since your endpoint is called /resolve you don't even need a status.